explorer/templates.go

```diff
@@ -194,6 +194,50 @@ func amountAsDecimalPartsTrimmed(v, numPlaces int64, useCommas bool) []string {
 	return []string{left, right, tail}
 }
 
+// threeSigFigs returns a representation of the float formatted to three
+// significant figures, with an appropriate magnitude prefix (k, M, B).
+// For (k, M, G) prefixes for file/memory sizes, use humanize.Bytes.
+func threeSigFigs(v float64) string {
+	if v >= 1e11 {
+		return fmt.Sprintf("%dB", int(math.Round(v/1e9)))
+	} else if v >= 1e10 {
+		return fmt.Sprintf("%.1fB", math.Round(v/1e8)/10)
+	} else if v >= 1e9 {
+		return fmt.Sprintf("%.2fB", math.Round(v/1e7)/1e2)
+	} else if v >= 1e8 {
+		return fmt.Sprintf("%dM", int(math.Round(v/1e6)))
+	} else if v >= 1e7 {
+		return fmt.Sprintf("%.1fM", math.Round(v/1e5)/10)
+	} else if v >= 1e6 {
+		return fmt.Sprintf("%.2fM", math.Round(v/1e4)/1e2)
+	} else if v >= 1e5 {
+		return fmt.Sprintf("%dk", int(math.Round(v/1e3)))
+	} else if v >= 1e4 {
+		return fmt.Sprintf("%.1fk", math.Round(v/100)/10)
+	} else if v >= 1e3 {
+		return fmt.Sprintf("%.2fk", math.Round(v/10)/1e2)
+	} else if v >= 100 {
```

Comment:
All the cases from here to the end can be covered by the magic of `"g"` where precision applies to the count of all digits, not just the ones to the right of the decimal:

```go
// either of:
strconv.FormatFloat(v, 'g', 3, 64)
fmt.Sprintf("%.3g", v)
```

Note that these functions round rather than truncate so the result should be as intended. https://play.golang.org/p/bgLV1WXNaxZ

Reply:
There seems to be some important differences. `'%g` will switch to scientific notation if is uses fewer characters. Also, `%g` removes trailing zeros, which can be undesirable.

Reply:
Good point about removing trailing zeros.  However, for the numbers below 100, and sig digits, it won't ever go to scientific notation.

Reply:
Hmm. Maybe I'm not understanding. https://play.golang.org/p/HUNZjKF-Tui

Reply:
Ah, got it.  Any smaller than 0.0001 and it switches to sci notation.